src/sentry/integrations/slack/endpoints/base.py

```diff
@@ -0,0 +1,99 @@
+from sentry.integrations.slack.requests.base import SlackRequest
+from sentry.integrations.slack.views.link_identity import build_linking_url
+from sentry.integrations.slack.views.unlink_identity import build_unlinking_url
+
+LINK_USER_MESSAGE = (
+    "<{associate_url}|Link your Slack identity> to your Sentry account to receive notifications. "
+    "You'll also be able to perform actions in Sentry through Slack. "
+)
+UNLINK_USER_MESSAGE = "<{associate_url}|Click here to unlink your identity.>"
+NOT_LINKED_MESSAGE = "You do not have a linked identity to unlink."
+ALREADY_LINKED_MESSAGE = "You are already linked as `{username}`."
+FEATURE_FLAG_MESSAGE = "This feature hasn't been released yet, hang tight."
+
+
+class SlackDMEndpoint(Endpoint, abc.ABC):  # type: ignore
+    def post_dispatcher(self, request: SlackRequest):
+        """
+        All Slack commands are handled by this endpoint. This block just
+        validates the request and dispatches it to the right handler.
+        """
+        command, args = self.get_command_and_args(request)
+
+        if command in ["help", ""]:
+            return self.respond(SlackHelpMessageBuilder().build())
+
+        integration = request.integration
+        organization = integration.organizations.all()[0]
+        if command in ["link", "unlink"] and not features.has(
+            "organizations:notification-platform", organization
+        ):
+            return self.reply(request, FEATURE_FLAG_MESSAGE)
+
+        if command == "link":
+            if not args:
+                return self.link_user(request)
+
+            if args[0] == "team":
+                return self.link_team(request)
+
+        if command == "unlink":
+            if not args:
+                return self.unlink_user(request)
+
+            if args[0] == "team":
+                return self.unlink_team(request)
+
+        # If we cannot interpret the command, print help text.
+        return self.respond(SlackHelpMessageBuilder(command).build())
+
+    def get_command_and_args(self, request):
+        raise NotImplementedError
+
+    def reply(self, slack_request, message: str) -> Response:
+        raise NotImplementedError
+
+    def link_user(self, slack_request):
+        if slack_request.has_identity:
+            return self.reply(
+                slack_request, ALREADY_LINKED_MESSAGE.format(username=slack_request.identity_str)
+            )
+
+        integration = slack_request.integration
+        organization = integration.organizations.all()[0]
+        associate_url = build_linking_url(
+            integration=integration,
+            organization=organization,
+            slack_id=slack_request.user_id,
+            channel_id=slack_request.channel_id,
+            response_url=slack_request.response_url,
+        )
+        return self.reply(slack_request, LINK_USER_MESSAGE.format(associate_url=associate_url))
+
+    def unlink_user(self, slack_request):
+        if not slack_request.has_identity:
```

Comment:
You need to implement `has_identity()` on `SlackEventRequest`.

Reply:
I'm confused, I did do that: https://github.com/getsentry/sentry/blob/d29a2a494c295485d2c678b4574f17adcedca49d/src/sentry/integrations/slack/requests/event.py#L35 

do you mean something else?

Reply:
Oops you're right. In `sentry/integrations/slack/endpoints/base.py` there is the line:
```python
def link_user(self, slack_request):
  if slack_request.has_identity:
    ...
```
It's actually `SlackRequest` that needs to have and un-implemented property.
```python
@property
def has_identity(self) -> bool:
  raise NotImplementedError
```